lua/fspectate/cl_init.lua

```diff
@@ -135,50 +208,30 @@ local function specCalcView()
 end
 
 --[[-------------------------------------------------------------------------
-Hitbox drawing code
+Chams drawing code
 ---------------------------------------------------------------------------]]
-local function drawGreenBoxes()
-    if not showHitboxes then return end
-    render.OverrideDepthEnable( true, false )
-
-    for _, v in ipairs( player.GetAll() ) do
-        if v == specEnt then continue end
-
-        for i = 0, v:GetHitBoxGroupCount() - 1 do
-            for _i = 0, v:GetHitBoxCount( i ) - 1 do
-                local bone = v:GetHitBoxBone( _i, i )
-                if not bone then continue end
-                local min, max = v:GetHitBoxBounds( _i, i )
-
-                if ( v:GetBonePosition( bone ) ) then
-                    local pos, ang = v:GetBonePosition( bone )
-                    render.DrawWireframeBox( pos, ang, min, max, Color( 0, 255, 0, 255 ) )
-                end
-            end
-        end
-    end
-
-    render.OverrideDepthEnable( false, false )
+local chamsmat1 = CreateMaterial( "CHAMSMATFSPEC1", "VertexLitGeneric", {["$basetexture"] = "models/debug/debugwhite", ["$model"] = 1, ["$ignorez"] = 1} )
+local chamsmat2 = CreateMaterial( "CHAMSMATFSPEC2", "VertexLitGeneric", {["$basetexture"] = "models/debug/debugwhite", ["$model"] = 1, ["$ignorez"] = 0} )
+local function drawCham( ply )
+    if not ply:Alive() then return end
+    if ply == specEnt and not thirdperson then return end
+
+    local col = team.GetColor( ply:Team() )
+    cam.Start3D();
+        render.SetColorModulation( col.r / 1000, col.g / 1000, col.b / 1000 )
+        render.MaterialOverride( chamsmat1 )
+        ply:DrawModel()
+        render.SetColorModulation( col.r / 255, col.g / 255, col.b / 255 )
+        render.MaterialOverride( chamsmat2 )
+        ply:DrawModel()
+        render.MaterialOverride()
+    cam.End3D();
```

Comment:
Given how frequently this runs and how important framerate is while recording, I think it'd be nice to get any savings we can.

```suggestion
local chamsmat2 = CreateMaterial( "CHAMSMATFSPEC2", "VertexLitGeneric", {["$basetexture"] = "models/debug/debugwhite", ["$model"] = 1, ["$ignorez"] = 0} )

local teamGetColor = team.GetColor
local Start3D = cam.Start3D
local SetColorModulation = render.SetColorModulation
local MaterialOverride = render.MaterialOverride
local End3D = cam.End3D

local function drawCham( ply )
    if not ply:Alive() then return end
    if ply == specEnt and not thirdperson then return end

    local col = ply._teamColor
    if not col then
        col = teamGetColor( ply:Team() )
        ply._teamColor = col
    end

    local r, g, b = col:Unpack()

    Start3D();
        SetColorModulation( r / 1000, g / 1000, b / 1000 )
        MaterialOverride( chamsmat1 )

        ply:DrawModel()

        SetColorModulation( r / 255, g / 255, b / 255 )
        MaterialOverride( chamsmat2 )

        ply:DrawModel()
        MaterialOverride()
    End3D();
```